libshaderc/include/shaderc.h

```diff
@@ -105,6 +111,11 @@ void shaderc_compile_options_release(shaderc_compile_options_t options);
 void shaderc_compile_options_add_macro_definition(
     shaderc_compile_options_t options, const char* name, const char* value);
 
+// Set flags for the rules on which warnings or errors will be issued. This can
+// either be an individual shaderc_msg_rules flag, or a set of flags or'd together
+void shaderc_compile_options_set_message_rules(
```

Comment:
Looking into the future once more, if we have two different calls to set target env and target env spec version, then we need to define when validation occurs. Since you can't set both at once, you can only validate at compile time, or we could add a "validate" method on the compiler options to check all settings for mutual consistency.